frame-metadata/src/v16.rs

```diff
@@ -223,22 +217,25 @@ impl IntoPortable for ExtrinsicMetadata {
 	serde(bound(serialize = "T::Type: Serialize, T::String: Serialize"))
 )]
 pub struct TransactionExtensionMetadata<T: Form = MetaForm> {
+	/// Extension version.
+	pub version: u8,
```

Comment:
In V5 transactions we'll have a single `u8` version denoting the _set_ of transaction extensions to use. So while we could use this approach and have eg:

```
vec![
   // v0 extensions
   TransactionExtensionMetadata { version: 0, identifier: "Foo", ty: 123, implicit: 456 },
   TransactionExtensionMetadata { version: 0, identifier: "Bar", ty: 124, implicit: 457 },
   // v1 extensions
   TransactionExtensionMetadata { version: 1, identifier: "Foo", ty: 123, implicit: 456 },
   TransactionExtensionMetadata { version: 1, identifier: "Bar", ty: 124, implicit: 457 },
   TransactionExtensionMetadata { version: 1, identifier: "Wibble", ty: 124, implicit: 457 },
   // ...
]
```

(noting that the order of extensions within each version is important)

It perhaps makes more sense to avoid the repetition and represent extensions something like:

```
// For each version number, list the indexes, in order, of the extensions used:
versions: Map<u8, Vec<u32>> {
    0: [0,1],
    1: [0,1,2]
},
// All of the extensions used. The above refers to these by index:
extensions: vec![
   TransactionExtensionMetadata { identifier: "Foo", ty: 123, implicit: 456 },
   TransactionExtensionMetadata { identifier: "Bar", ty: 124, implicit: 457 },
   TransactionExtensionMetadata { identifier: "Foo", ty: 123, implicit: 456 },
   TransactionExtensionMetadata { identifier: "Bar", ty: 124, implicit: 457 },
   TransactionExtensionMetadata { identifier: "Wibble", ty: 124, implicit: 457 },
]
```